src/ppl/src/IOPlacer.cpp

```diff
@@ -709,7 +741,54 @@ void IOPlacer::findSlots(const std::set<int>& layers, Edge edge)
 
     for (const Point& pos : slots) {
       bool blocked = checkBlocked(edge, pos, layer);
-      slots_.push_back({blocked, false, pos, layer, edge});
+      // For regular die boundary edges, use default-constructed line
+      odb::Line empty_line;  // Default constructor creates empty line
+      slots_.push_back({blocked, false, pos, layer, edge, empty_line});
+    }
+  }
+}
+
+void IOPlacer::findSlotsPolygon(const std::set<int>& layers, odb::Line line)
```

Comment:
This function should be merged with `findSlots` too. The differences between them are pretty minor, and could follow the same logic you'll apply for the other functions that will be merged.